provider/mqtt/client.go

```diff
@@ -75,9 +79,24 @@ func NewClient(log *util.Logger, broker, user, password, clientID string, qos by
 	options.SetConnectionLostHandler(mc.ConnectionLostHandler)
 	options.SetConnectTimeout(request.Timeout)
 
-	if insecure {
-		options.SetTLSConfig(&tls.Config{InsecureSkipVerify: true})
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: insecure,
```

Comment:
Should this still be allowed if certficates are specified or are these mutually exclusive?

Reply:
In theory they are, because the provided certificate is not used to verify the servers identity (this is only for the Root-CA option). However you could set a custom certificate and still tell the client to ignore said certificate, eg. to get a client set up but the server doesn't have the certificate yet. Later you'd only need to disable the insecure flag. A niche use case but I'd let the user decide on this.